test/tcp_echo_client/tcp_echo_client.cpp

```diff
@@ -20,17 +20,201 @@
+ * ARGUMENTS:
+ *  description     client string to provide contextural information for
+ *                  the ipaddr:port tuple
+ *  addr            ip address data structure
+ *  port            port number
+ * RETURN:
+ *  None
+ *****************************************************************************/
+static inline void tcp_ec_sock_addr_port_dump(const char* description, struct socket_addr* saddr, uint16_t port)
+{
+    if(saddr)
+    {
+        TEST_PRINT("%s:%s:%d\r\n", description, inet_ntoa(*saddr), port);
+    }
+}
+
+/*****************************************************************************
+ * FUNCTION: tcp_ec_client_cb
+ * callback function for handling tx/rx/etc event indications.
+ * script
+ * ARGUMENTS:
+ *  None
+ * RETURN:
+ *  None
+ *****************************************************************************/
+static void tcp_ec_client_cb()
+{
+    struct socket_event *e = tcp_ec_client_socket_g->event;
+    event_flag_t event = e->event;
+    switch (event) {
+        case SOCKET_EVENT_RX_DONE:
+            tcp_ec_client_rx_done_g = true;
+            break;
+        case SOCKET_EVENT_TX_DONE:
+            tcp_ec_client_tx_done_g = true;
+            tcp_ec_client_tx_info_g.sentbytes = e->i.t.sentbytes;
+            break;
+        case SOCKET_EVENT_DISCONNECT:
+            tcp_ec_closed = 1;
+            break;
+        case SOCKET_EVENT_CONNECT:
+            tcp_ec_connected = 1;
+            break;
+        default:
+            memcpy((void *) &tcp_ec_client_event_g, (const void*)e, sizeof(e));
+            tcp_ec_client_event_done_g = true;
+            break;
+    }
+}
+
+/*****************************************************************************
+ * FUNCTION: tcp_ec_send_shutdown_host_script
+ * send a shutdown command to the host test script to terminate host pc test
+ * script
+ * ARGUMENTS:
+ * srv_addr_s   dotted decimal ip addr string of the remote tcp echo server
+ * srv_port     port number of the remote tcp echo relay server
+ * RETURN:
+ *  0 => success, !=0 => failure.
+ *****************************************************************************/
+static int tcp_ec_send_shutdown_host_script(const char* srv_addr_s, uint16_t srv_port)
+{
+    struct socket s;
+    socket_error_t err;
+    struct socket_addr srv_saddr = {0, 0, 0, 0};
+    struct socket_addr local_saddr = {0, 0, 0, 0};
+    const struct socket_api *api = socket_get_api(SOCKET_STACK_LWIP_IPV4);
+    mbed::Timeout to;
+    const char* shutdown_cmd = "shutdown";
+    uint16_t local_port = 0;
+
+    TEST_CLEAR();
+    tcp_ec_client_socket_g = &s;
+
+    /* Create the socket */
+    err = api->init();
+    if (!TEST_EQ(err, SOCKET_ERROR_NONE)) {
+        TEST_RETURN();
+    }
+    s.impl = NULL;
+    err = api->create(&s, SOCKET_AF_INET4, SOCKET_STREAM, &tcp_ec_client_cb);
+    if (!TEST_EQ(err, SOCKET_ERROR_NONE))
+    {
+        TEST_EXIT();
+    }
+    err = api->bind(&s, &local_saddr, local_port);
```

Comment:
Understood, but I was under the impression that the former was automatic in the latter. Regardless, this *should* work, so it's a valid test case.